apps/bfd-pipeline/bfd-pipeline-rif-load/src/main/java/gov/cms/bfd/pipeline/rif/load/RifLoaderIdleTasks.java

```diff
@@ -0,0 +1,355 @@
+   * Create a helper to manage the idle time tasks.
+   *
+   * @param options pipeline options
+   * @param appMetrics pipeline metrics
+   * @param entityManagerFactory a connection to the database of the pipeline
+   * @param secretKeyFactory for hashing
+   */
+  public RifLoaderIdleTasks(
+      final LoadAppOptions options,
+      final MetricRegistry appMetrics,
+      final EntityManagerFactory entityManagerFactory,
+      final SecretKeyFactory secretKeyFactory) {
+    this.options = options;
+    this.entityManagerFactory = entityManagerFactory;
+    this.secretKeyFactory = secretKeyFactory;
+
+    this.beneficaryMeter = appMetrics.meter("fixups.beneficiary.rate");
+    this.historyMeter = appMetrics.meter("fixups.beneficiary_history.rate");
+
+    this.executorService = Executors.newFixedThreadPool(THREAD_COUNT);
+  }
+
+  /**
+   * What task will be executed in the next idle time slot
+   *
+   * @return the current task
+   */
+  public Task getCurrentTask() {
+    return currentTask;
+  }
+
+  /**
+   * Run the current idle task. This method is expected to be called whenever no RIF files are
+   * present for process. It will respect the TASK_TIME_LIMIT to allow checking of RIF files and to
+   * not interfer with RIF file processing.
+   */
+  public void doIdleTask() {
+    switch (currentTask) {
+      case INITIAL:
+        if (doInitialTask()) {
+          currentTask = Task.POST_STARTUP;
+        }
+        break;
+      case POST_STARTUP:
+        if (doPostStartupTask()) {
+          currentTask = Task.NORMAL;
+        }
+        break;
+      case NORMAL:
+        if (doNormalTask()) {
+          currentTask = Task.NORMAL;
+        }
+        break;
+      default:
+        throw new RuntimeException("Unexpected idle task");
+    }
+  }
+
+  /**
+   * Run this task as the first idle task.
+   *
+   * @return true if done with this task.
+   */
+  public boolean doInitialTask() {
+    // For the log count the work that we have to do.
+    final EntityManager em = entityManagerFactory.createEntityManager();
+    final Long beneficiaryCount =
+        em.createQuery(COUNT_UNHASHED_BENFICIARIES, Long.class).getSingleResult();
+    final Long historyCount =
+        em.createQuery(COUNT_UNHASHED_HISTORIES, Long.class).getSingleResult();
+
+    LOGGER.info(
+        "Starting idle task processing with null mbiHash for: {} Beneficaries and {} Benficiary Histories",
+        beneficiaryCount,
+        historyCount);
+
+    return true;
+  }
+
+  /**
+   * Run this task after the initial task. Respect the TASK_TIME_LIMIT.
+   *
+   * @return true if done with the current task.
+   */
+  public boolean doPostStartupTask() {
+    final Instant startTime = Instant.now();
+    LOGGER.debug("Started a PostStartup time slice");
+
+    // Execute batches in parallel
+    List<Future<Boolean>> tasks = new ArrayList<Future<Boolean>>();
+    tasks.add(executorService.submit(() -> doTransaction(startTime, this::fixupBeneficiaryBatch)));
```

Comment:
Possibly change this `tasks` variable to something like `jobs`, I found it a bit confusing differententiating between these `tasks` and tasks like `doPostStartupTask`. Particularly in the `waitUntilDone` function.

Reply:
I'm not sure I follow. Let's talk.